trace_decoder/src/decoding.rs

```diff
@@ -618,6 +628,47 @@ fn process_txn_info(
     Ok(gen_inputs)
 }
 
+fn merge_batch_jumpdest_tables<'t, T>(jdts: T) -> Option<JumpDestTableWitness>
+where
+    T: Iterator<Item = &'t Option<JumpDestTableWitness>>,
+{
+    let mut merged_table = JumpDestTableWitness::default();
+
+    let mut max_batch_ctx = 0;
+    for jdt in jdts {
+        let tx_offset = max_batch_ctx;
+        // abort if any transaction in the batch came without RPC JumpDestTable.
+        if jdt.is_none() {
+            return None;
+        }
```

Comment:
I left a TODO.

Out of curiosity: Would we have enough state available to continue the execution (of the batch) where this tx starts?

Reply:
What do you mean by `enough state available` here?

At a very high level (I didn't really think the details through), the idea is that because the execution on the proving side is sequential, similarly to how JDTs are being prefetched, we shouldn't in theory need a mapping with context.
When entering a new context after loading a user bytecode, the prover (within `prover_input`) would either do nothing if the codehash does not match the top element of the packed preprocessed JDTs, or pop this table and load the associated proofs.

Reply:
I was thinking of the interpreter's state from which it can continue, but maybe I am missing the bigger picture.